components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/interceptors/RESTAPISecurityInterceptor.java

```diff
@@ -118,6 +119,15 @@ public boolean preCall(Request request, Response response, ServiceMethodInfo ser
                 response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(Json.pretty
                         (swagger)).setMediaType(MediaType.APPLICATION_JSON).send();
                 return false;
+            } else if (requestURI.contains("swagger.yaml")) {
+                try {
+                    yamlContent = RestApiUtil.getStoreRestAPIResource();
+                } catch (APIManagementException e) {
+                    log.error("Couldn't find swagger.yaml for store", e);
+                }
+                response.setStatus(javax.ws.rs.core.Response.Status.OK.getStatusCode()).setEntity(yamlContent)
```

Comment:
If the line RestApiUtil.getStoreRestAPIResource() throws an Exception, the response should be sent with an appropriate error code.